### PR TITLE
Add azureblob receiver and googlecloudpubsub receiver

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Adds two receivers for log use cases: Azure blob receiver and Google Cloud PubSub receiver.

### CODE REVIEW

File: docs/components.md [hunk 1]
```diff
@@ -26,6 +27,7 @@ The distribution offers support for the following components.
 | [elasticsearch](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/elasticsearchreceiver)                   | [beta]           |
 | [filelog](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver)                               | [beta]           |
 | [fluentforward](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/fluentforwardreceiver)                   | [beta]           |
+| [googlecloudpubsub]()                                                                                                                         | [beta]           |
```
Comment (atoulme) [inline docs/components.md:30]:
```suggestion
| [googlecloudpubsub](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/googlecloudpubsubreceiver)    | [beta]           |
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+### 🚀 New components 🚀
+
+- (Splunk) Add Azure Blob receiver ([#](https://github.com/signalfx/splunk-otel-collector/pull/))
+- (Splunk) Add Google Cloud PubSub receiver ([#](https://github.com/signalfx/splunk-otel-collector/pull/))
```
Comment (atoulme) [inline CHANGELOG.md:8]:
```suggestion
- (Splunk) Add Azure Blob receiver ([#5200](https://github.com/signalfx/splunk-otel-collector/pull/5200))
- (Splunk) Add Google Cloud PubSub receiver ([#5200](https://github.com/signalfx/splunk-otel-collector/pull/5200))
```

File: docs/components.md [hunk 3]
```diff
@@ -17,6 +17,7 @@ The distribution offers support for the following components.
 | [awscontainerinsights](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/awscontainerinsightreceiver)      | [beta]           |
 | [awsecscontainermetrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/awsecscontainermetricsreceiver) | [beta]           |
 | [apache](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/apachereceiver)                                 | [alpha]          |
+| [azureblob](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/azureeventhubreceiver)                       | [alpha]          |
```
Comment (crobert-1) [inline docs/components.md:20]:
```suggestion
| [azureblob](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/azureblobreceiver)                       | [alpha]          |
```
_Note: My suggestion may break spacing in the table, I'm not sure._

File: docs/components.md [hunk 4]
```diff
@@ -17,6 +17,7 @@ The distribution offers support for the following components.
 | [awscontainerinsights](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/awscontainerinsightreceiver)      | [beta]           |
 | [awsecscontainermetrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/awsecscontainermetricsreceiver) | [beta]           |
 | [apache](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/apachereceiver)                                 | [alpha]          |
+| [azureblob](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/azureblobreceiver)                       | [alpha]          |
```
Comment (atoulme) [inline docs/components.md:20]:
```suggestion
| [azureblob](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/azureblobreceiver)                           | [alpha]          |
```

File: docs/components.md [hunk 5]
```diff
@@ -26,6 +27,7 @@ The distribution offers support for the following components.
 | [elasticsearch](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/elasticsearchreceiver)                   | [beta]           |
 | [filelog](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver)                               | [beta]           |
 | [fluentforward](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/fluentforwardreceiver)                   | [beta]           |
+| [googlecloudpubsub](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/googlecloudpubsubreceiver)    | [beta]           |
```
Comment (atoulme) [inline docs/components.md:30]:
```suggestion
| [googlecloudpubsub](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/googlecloudpubsubreceiver)           | [beta]           |
```

**Pull request merged**